tests/integration/test_repeat.py

```diff
@@ -23,42 +22,183 @@
 np.random.seed(12345)
 
 
-def test_basic():
-    assert np.array_equal(num.repeat(3, 4), np.repeat(3, 4))
-    assert np.array_equal(num.repeat([3, 1], 4), np.repeat([3, 1], 4))
+@pytest.mark.parametrize(
+    "array", (None, [], 4, [2, 3], mk_seq_array(num, (3, 4, 2)))
+)
+def test_repeats_none(array):
+    with pytest.raises(TypeError):
+        num.repeat(array, None)
+
+
+@pytest.mark.parametrize("repeats", (-3, [], [-3], [2, 3]))
+def test_array_none_invalid(repeats):
+    with pytest.raises(ValueError):
+        num.repeat(None, repeats)
+
+
+@pytest.mark.parametrize("repeats", (3, [0], [3], 4.7, [4.7]))
+def test_array_none_valid(repeats):
+    res_num = num.repeat(None, repeats)
+    res_np = np.repeat(None, repeats)
+    assert np.array_equal(res_np, res_num)
+
+
+@pytest.mark.parametrize("repeats", (-3, 0, 3, 4.7, [], [-3], [0], [3], [4.7]))
+def test_array_empty_repeats_valid(repeats):
+    res_np = np.repeat([], repeats)
+    res_num = num.repeat([], repeats)
+    assert np.array_equal(res_np, res_num)
+
+
+@pytest.mark.parametrize("repeats", ([3, 4], [1, 2, 3]))
+def test_array_empty_repeats_invalid(repeats):
+    # numpy raises:
+    # ValueError: operands could not be broadcast together with shape (0,) (2,)
+    # while cunumeric is pass with the result []
+    res_num = num.repeat([], repeats)
+    assert np.array_equal(res_num, [])
```

Comment:
@xialu00 if there are cases where cunumeric and numpy disagree, we have also started adding an `xfail` test (that asserts equality) to help document the discrepancy and make it more visible. Here is an example from a other PR:

https://github.com/nv-legate/cunumeric/pull/658/files#diff-3a50b1bf66b44299a86b1a22d62f8ae944642d341405b278be5395c99caf17e9R108-R115

Reply:
just to clarify, we should keep the positive test above, but add an `xfail` test in addition.

Reply:
Got it. Will mark it as _xfail_ if the same situation occurs later.

Reply:
@xialu00 it looks like you removed the previous test? We need both:

* A regular test (not marked `xfail`) that maintains what the cunumeric behavor actually is (whatever it is) with positive assertions for actual values
* We need an `xfail` test the maintains and documents the discrepancy with numpy (by asserting equality with numpy)